src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,37 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /*
+   * Implemented by means of a buffer to keep track of the last n elements during iteration.
+   */
+  def takeRight(n: Int): Iterator[A] = {
+    if (n == 0) Iterator.empty
+    else {
+      val buffer = ArrayBuffer[A]()
+      var count = 0
+      var index = 0
+      // First iterate over all elements while keeping track of the last n
+      while (self.hasNext) {
+        if (index >= buffer.length) buffer += self.next()
+        else buffer(index) = self.next()
+        index = if ((index + 1) >= n) 0 else index + 1
+        if (count < n) count += 1
+      }
+      // Adjust the starting index if needed
+      index = if (index >= buffer.length) 0 else index
+      // Return an iterator for the elements in the buffer starting from index
+      new Iterator[A]() {
+        override def hasNext: Boolean = count > 0
+        override def next(): A = {
+          val value = buffer(index)
+          index = if (index + 1 >= buffer.length) 0 else index + 1
```

Comment:
...then you wouldn't need another copy of it here and could simply do `index += 1`